File: Python/lab8/Interface/Product.py
from abc import ABC, abstractmethod
from Interface.RepairStages import RepairStages

class Product(ABC):
    def __init__(self, repair_stages: RepairStages):
        self.repair_stages = repair_stages

    def get_repair_stages(self):
        return self.repair_stages

    @abstractmethod
    def get_info_about_stage(self):
        pass

File: Java/lab8/src/main/java/org/example/Interface/Product.java
package org.example.Interface;
public abstract class Product {
    protected RepairStages repairStages;
    public RepairStages getRepairStages() {
        return repairStages;
    }

    public void setRepairStages(RepairStages repairStages) {
        this.repairStages = repairStages;
    }

    public Product(RepairStages repairStages) {
        this.repairStages = repairStages;
    }

    public abstract void getInfoAboutStage();
}
